### ISSUE TITLE
Crash when I map axis from multiple joysticks

### ISSUE THREAD

**Issue by NdranC** [Snoothy/UCR#98]
I have been using UCR for a while only to merge the toe breaks on my pedals. I wanted to start using it to control the sensitivity of both my T16000M joysticks (dual joystick + pedals setup) and I get constant crashes when I'm trying to use it or set it up. I managed to finish the setup but it seems like there is a very high chance it will crash the moment I start using the axis of any of them. Haven't been able to pin point exactly what seems to be the trigger for the crash except that maybe it has to do with the fact that it's 2 joysticks of the same model and that might be confusing it?.

As it stands my config is Right Joystick to vJoy1, Left Joystick to vJoy2 and Pedals to vJoy3.

This is the crash log:

[2019-05-17.log](https://github.com/Snoothy/UCR/files/3200058/2019-05-17.log)

EDIT: Just to clarify it seems the logs don't reflect anything wrong for the most part. The program just basically slowly hangs and the inputs stop working. I have only been able to successfully use it with the pedals and nothing else. I'll try to keep adding one axis at a time and test it that way.

**Comment by Snoothy** [comment]
Please try again with release v0.8.0

**Comment by NdranC** [comment]
I did. I  was able to register all the mappings to vJoy, 
T16000M (Right) x+y+z > vJoy1 x+y+z
T16000M (Left) x+y+z > vJoy2 x+y+z
T-Rudder (x+y+z) > vJoy3 x+y+z (x and y merged)
The program becomes unstable, it crashes often either immediately when I hit run profile or when I start using any of the axis more often.

I have no issues running this setup with the old UCR_v0.1.22

I still have it installed but besides the general crash-iness I have no idea how else to contribute. Is there anything I can do to provide more info?

**Comment by Snoothy** [comment]
Please upload your context.xml from the v0.8.0 version

**Comment by evilC** [comment]
I think it is more likely to be due to multiple sticks rather than identical sticks. Each stick has it's own unique ID, so once actually polling sticks, DirectInput does not care if they are identical.  
I have a T16k / FLCS combo, plus a 3DRudder which can operate in DI mode, so I will see if I can repro any issues with 3 sticks.  
If you have another stick which is not identical though, would probably be worth checking to see if you can repro with that.  
Looking at the crash log, it seems to happen in the `Dispose` method, which is called when something is no longer needed - this is most odd. Was the crash in this log when setting up (eg using Bind Mode), or when the profile was in use?

**Comment by evilC** [comment]
PS, if you join the [Discord channel](https://discord.gg/MmnhQYQ), I can maybe try giving you some updated files to try and see if we can fix this crash

**Comment by evilC** [comment]
PPS, I just posted a replacement DLL to the Discord channel which may fix the crash. I also see a potential reason why the Dispose method is getting called - if there is an error when polling a stick, then it will be disposed, so I am thinking this fix may just shunt the problem elsewhere.

**Comment by NdranC** [comment]
Sorry had to step away from the computer, here is the context.xml in pastebin:
https://pastebin.com/Tnsqjbgh

I don't have a lot of time to do debug right this second but I might be able soon-ish. In the mean time if you need anything else I can provide quickly let me know.

**Comment by NdranC** [comment]
Just tried the dll posted on discord. I seem to be able to replicate the crash fairly often by starting and stopping the profile several times. Usually crashes when I try to stop at some point. I don't have the time right this second to test it over a long period of use but I might be able this afternoon. I'll report what I find.

**Comment by evilC** [comment]
I now know where it is going boom in IOWrapper (The UCR back end), but I am not sure why...  
https://github.com/evilC/IOWrapper/issues/33

**Comment by Snoothy** [comment]
Is this still an issue with the v0.9.0 release?

**Comment by will-thethrill** [comment]
Yeah. I'm having the same issue in v0.9.0 and the dev v0.11.3. Please help...

**Comment by dylan1x** [comment]
I'm having this issue on v0.9.0 also.
